.github/workflows/build.yml

```
@@ -0,0 +1,95 @@
+name: Dolphin-AppImage
+
+on: 
+  push:
+
+jobs:
+  dolphin-appimage:
+    name: dolphin bundled as appimage
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Runs the Dolphin action
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+      - name: cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.ccache
+          key: build-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            build-ccache-
+      - name: Install Packages
+        continue-on-error: true
+        id: packages
+        run: |
+          sudo add-apt-repository -y ppa:beineri/opt-qt-5.15.2-focal
+          sudo apt-get install -y dialog libbluetooth-dev liblz4-dev libopus-dev libsystemd-dev \
+          libzip-dev libzstd-dev libasound2-dev libpulse-dev pulseaudio libsfml-dev libminiupnpc-dev libmbedtls-dev libpugixml-dev \
+          liblzo2-dev qt515base qt5-qmake libxi-dev libavcodec-dev libudev-dev libusb-1.0-0-dev libevdev-dev libhidapi-dev libavformat-dev libavdevice-dev \
+          libfmt-dev libwayland-dev libxrandr-dev libglu1-mesa-dev libcurl4-openssl-dev zenity gettext ccache cmake ninja-build \
+      - name: Install Packages if failed
+        id: packages2
+        if: steps.packages.outcome=='failure'
+        run: |
+          sudo add-apt-repository -y ppa:beineri/opt-qt-5.15.2-focal
+          sudo apt-get install -y build-essential dialog dpkg libbluetooth-dev liblz4-dev liblzma-dev libssl-dev libopus-dev libpng-dev libsystemd-dev \
+          libzip-dev libzstd-dev zlib1g-dev libasound2-dev libpulse-dev pulseaudio p7zip p7zip-full libsfml-dev libminiupnpc-dev libmbedtls-dev libpugixml-dev \
```

`p7zip` is a dependency of `p7zip-full` (there are probably other redundant packages but I cba to go through them)
```suggestion
          libzip-dev libzstd-dev zlib1g-dev libasound2-dev libpulse-dev pulseaudio p7zip-full libsfml-dev libminiupnpc-dev libmbedtls-dev libpugixml-dev \
```